[new product] add Coder

matifali opened the pull request:
Adds Coder release schedule as per https://coder.com/docs/install/releases#release-schedule
Closes #5482

matifali committed:
add coder release schedule

based on https://coder.com/docs/install/releases#release-schedule

matifali committed:
bumo 2.12 version

marcwrobel committed:
Rename coder.yaml to coder.md

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,80 @@
+
+---
+
+> [Coder](https://github.com/coder/coder) is an open-source cloud development environment (CDE) that you host in your cloud or on-premises. With Coder, you can deploy environments that provide the infrastructure, IDEs, and tools your developers need. Upgrade to Coder Enterprise to gain enhanced security, governance, and observability for your platform teams.
```
```suggestion
> [Coder](https://coder.com) is an open-source platform for creating and managing developer workspaces on your preferred
> clouds and servers.
```

Shorter description from https://coder.com/docs.

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,80 @@
+title: Coder
+category: server-app
+tags: remote cde terraform ide development docker kubernetes
```
```suggestion
tags: build-tool
```

Only a limited set of tags is accepted, see https://endoflife.date/recommendations. The closest I could find is `build-tool` for now (we don't have an `ide` nor a `developer-tool` tag). WDYT ?

matifali replied:
I think. It doesn't come under a `build-tool` what do you think about supporting a new tag like `ide` or/and `development`?

marcwrobel replied:
Not for now, this should be done in a separate PR if there is enough candidate (we are not adding new tags unless it is the vendor name or applicable to at least three products (see https://endoflife.date/contribute, previously shared the wrong link).

So for now I will just delete the tags (`app` is a category and cannot be used as a tag).

matifali replied:
I see app being used as a tag in many apps. 
`server-app` is a category.

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,80 @@
+title: Coder
```
```suggestion
---
title: Coder
```

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,80 @@
+eolColumn: Stable
+eolWarnThreshold: 60
+eoesColumn: Security Support
+eoasWarnThreshold: 150
```
```suggestion
eoesWarnThreshold: 150
```

Did you mean this ?.

matifali replied:
Yes. Thanks for catching the typo.

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,80 @@
+  eol: 2024-09-03
+  eoes: false
+  latest: "2.13.1"
+  latestReleaseDate: 2024-07-20
```
```suggestion
  latest: "2.13.2"
  latestReleaseDate: 2024-07-23
```

matifali replied:
Thank you. I wanted to bump this version.

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,80 @@
+  eol: 2024-07-02
+  eoes: false
+  latest: "2.11.3"
+  latestReleaseDate: 2024-06-21
```
```suggestion
  latest: "2.11.4"
  latestReleaseDate: 2024-06-24
```

matifali replied:
Thank you.

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,80 @@
+  eol: 2024-04-03
+  eoes: true
+  latest: "2.8.4"
+  latestReleaseDate: 2024-03-09
```
```suggestion
  latest: "2.8.5"
  latestReleaseDate: 2024-03-09
```

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,80 @@
+iconSlug: coder
+permalink: /coder
+versionCommand: coder version
+releasePolicyLink: https://coder.com/docs/install/releases#release-schedule
```
```suggestion
releasePolicyLink: https://coder.com/docs/install/releases
```

All the page is interesting IMO.

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,80 @@
+
+---
+
+> [Coder](https://github.com/coder/coder) is an open-source cloud development environment (CDE) that you host in your cloud or on-premises. With Coder, you can deploy environments that provide the infrastructure, IDEs, and tools your developers need. Upgrade to Coder Enterprise to gain enhanced security, governance, and observability for your platform teams.
```
The product needs a description of the release and support policy. From what I understood:

- the last three versions are supported
- there are three stages of support:
  - mainline: bug and security fixes
  - stable: major bug and security fixes
  - security: only security fixes

Unfortunately endoflife.date structure is not suited for this kind of support. You used the `eoesColumn` for documenting security support but this does not fit : extended support should be reserved for support past EOL, such as commercial support. So given the security phase is a standard support phase, it should be documented in the `eolColumn`.

I propose to document both mainline and stable versions in the active support. I agree it's not ideal, but it's close to reality and best fit with endoflife.date way of documenting versions.

WDYT ?

marcwrobel replied:
And given what I understood of the support policy, we could also apply the following rule:

- eoas(x) = releaseDate(x+2)
- eol(x) = releaseDate(x+3)

matifali replied:
This sounds nice. I would appreciate some help on how can we apply these rules?

matifali replied:
> Unfortunately endoflife.date structure is not suited for this kind of support. You used the eoesColumn for documenting security support but this does not fit : extended support should be reserved for support past EOL, such as commercial support. So given the security phase is a standard support phase, it should be documented in the eolColumn.

I agree.

matifali replied:
> I propose to document both mainline and stable versions in the active support. I agree it's not ideal, but it's close to reality and best fit with endoflife.date way of documenting versions.

I agree and we do support both of them. Its just that once a release is cut it does not get new features but only bug and security fixes.

matifali replied:
I think I have fixed it for now. And I like the current preview. I hope it will be automated.

marcwrobel replied:
Unfortunately for now EOAS / EOL cannot be automated unless there is a page on the official website documenting the dates.

marcwrobel reviewed (changes requested):
Thanks for this contribution @matifali, you can find my remarks below.

Regarding automation: 
> I would appreciate a review on how to optimize and automate this entry further. I reviewed the contributing guide and tried to make it follow the guidelines.

We could use the on https://coder.com/docs/install/releases#release-schedule if it does not match the tag date of the first version (did not checked). But I don't see any way to automate further as release statuses is documented as text (Not Supported, Security support...).

marcwrobel committed:
Update products/coder.md

matifali committed:
Apply suggestions from code review

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

matifali committed:
remove eoes and update dates

apply rule
# eoas(x) = releaseDate(x+2)
# eol(x) = releaseDate(x+3)

matifali committed:
replace dates with `false` for unreleased versions.

matifali committed:
remove eoas warning

matifali committed:
change category to `app`

I think `app` should be the best category given other provisioning tools like `terraform` and `ansible` are using this. 

I would still prefer a tag `ide` or/and `development`.

matifali committed:
fixup!

matifali committed:
add eoas column

matifali committed:
enable warning on eoas column

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,78 @@
+---
+title: Coder
+category: server-app
+tags: app
```
```suggestion
```

Categories cannot be used a a tag.

marcwrobel committed:
Update products/coder.md

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,77 @@
+releasePolicyLink: https://coder.com/docs/install/releases
+changelogTemplate: https://github.com/coder/coder/releases/tag/v__LATEST__
+releaseDateColumn: true
+eoasColumn: Active Support
```
```suggestion
```

Already the default value.

matifali replied:
Without this, the column is hidden and does not show up.

marcwrobel replied:
Oups, you're right, sorry.

matifali replied:
I wanted to add this column. I will submit a follow up PR.

marcwrobel replied:
Done in #5590.

marcwrobel commented on products/coder.md:
```diff
@@ -0,0 +1,77 @@
+---
+
+> [Coder](https://coder.com) is an open-source platform for creating and managing developer workspaces on your preferred
+> clouds and servers.
```
```suggestion
> clouds and servers.

There is a new minor release of Coder on the first Tuesday of each month. Minor releases are
supported for three months with bug and security fixes the first month, major bug and security
fixes the second month and only security fixes the third month.
```

WDYT of this description ?

matifali replied:
Looks good for now and matches the release policy exactly. Thank you.

marcwrobel committed:
Update products/coder.md

matifali committed:
Merge branch 'endoflife-date:master' into patch-2

matifali committed:
address PR comments

marcwrobel reviewed (approved):
Quickly rechecked https://deploy-preview-5496--endoflife-date.netlify.app/coder, LGTM, merging. Thanks a lot for this contribution @matifali.
